app/client/components/cancel/cancellationFlow.tsx

```diff
@@ -53,6 +54,9 @@ interface ReasonPickerState {
   cancellationPolicy?: string;
 }
 
+const shouldShow = (reason: CancellationReason, productDetail: ProductDetail) =>
+  reason.shouldShow ? reason.shouldShow(productDetail) : true;
+
```

Comment:
Seemed easiest to make `shouldShow` optional and use it like this, rather than add e.g

```js
shouldShow = () => true
```

to all the  cancellation reasons